packages/hardhat-core/src/internal/hardhat-network/provider/provider.ts

```diff
@@ -133,101 +123,48 @@ export class HardhatNetworkProvider extends EventEmitter
       //  fails without throwing, this will be displayed in green. It's unclear
       //  if this is correct. See Eth module's TODOs for more info.
 
-      if (this._shouldCollapseMethod(method)) {
-        this._logCollapsedMethod(method);
-      } else {
-        this._startCollapsingMethod(method);
-        this._log(method, false, chalk.green);
-      }
+      if (method !== "hardhat_intervalMine") {
+        this._logger.printMethod(method);
 
-      const loggedSomething = this._logModuleMessages();
-      if (loggedSomething) {
-        this._stopCollapsingMethod();
-        this._log("");
+        const printedSomething = this._logger.printLogs();
+        if (printedSomething) {
+          this._logger.printEmptyLine();
+        }
       }
 
       return result;
     } catch (err) {
-      this._stopCollapsingMethod();
-
       if (
         err instanceof MethodNotFoundError ||
         err instanceof MethodNotSupportedError
       ) {
-        this._log(`${method} - Method not supported`, false, chalk.red);
+        this._logger.printMethodNotSupported(method);
 
         throw err;
       }
 
-      this._log(method, false, chalk.red);
+      this._logger.printFailedMethod(method);
+      this._logger.printLogs();
 
-      const loggedSomething = this._logModuleMessages();
-      if (loggedSomething) {
-        this._log("");
-      }
-
-      if (err instanceof LoggedError) {
-        this._log("");
-        throw err.wrappedError;
-      }
-
-      if (err instanceof SolidityError) {
-        this._logError(err);
-      } else if (err instanceof HardhatNetworkProviderError) {
-        this._log(err.message, true);
+      if (err instanceof HardhatNetworkProviderError) {
```

Comment:
This is the tricky part. `HardhatNetworkProviderError` is pretty wide. It includes, for example, `TransactionExecutionError`, which is used for OOG errors. 

I'd at least ignore `TransactionExecutionError`. This will make the test with the comment work as expected/before.